receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -28,6 +28,14 @@ import (
 )
 
 func newRemoteWriteReceiver(settings receiver.Settings, cfg *Config, nextConsumer consumer.Metrics) (receiver.Metrics, error) {
+	if settings.BuildInfo.Version != "" {
+		buildVersion = settings.BuildInfo.Version
+	}
+
+	if settings.BuildInfo.Description != "" {
+		buildName = settings.BuildInfo.Description
+	}
+
 	return &prometheusRemoteWriteReceiver{
```

Comment:
Regardless of whether `settings.BuildInfo.Version` or `settings.BuildInfo.Description` are empty, we should always assign their values to buildName and version. The determining factor for whether we use them or not will be whether scope_name or version are empty. I think that we can simplify the `ifs` here.

```suggestion
	buildVersion = settings.BuildInfo.Version

    buildName = settings.BuildInfo.Description

	return &prometheusRemoteWriteReceiver{
```

Reply:
True :)

We don't even need the extra variables, actually. We can update the function like this:

```go
func (prwr * prometheusRemoteWriteReceiver) addGaugeDatapoints(..... {
  scopeName := prwr.settings.BuildInfo.Description
  scopeVersion := prwr.settings.BuildInfo.Version
}
```

If the settings are empty, the scope variables will be empty. If settings had proper content, then the scope will be populated correctly